### PR TITLE
Show "Add Option" button for only users who have permission

### LINKED ISSUE #288: Default behavior and buttons don't seem to match options
#### Summary
The default addionnal buttons specifically the Àdd Option` one is always on no matter if the `--public-add-option` is used or not.

#### Steps to reproduce
1. start a poll with `/poll "q" "a" "b"`
2. the following popup shows up for **ALL** users
image

I wouldn't want the users to be able to add an option by default and i didn't use the  `--public-add-option` option. 

#### Expected behavior
Users shouldn't see that button

#### Observed behavior (that appears unintentional)
Add option button shows up

#### Server configuration
**Operating system:**
Linux

**Mattermost version:**
5.21.0

**Matterpoll version:**
1.3.0

**Updated from an older Matterpoll version or fresh install:**
Fresh Install

### PR DESCRIPTION
**Pull request by kaakaa**
## Summary
This PR makes "Add Option" button visible for only users who have permission to add options. "Add Option" button is shown for the following users.

* system administrator
* poll creator
* (only if `--public-add-option` set) all users

スクリーンショット 2020-04-29 9 54 25

Fixes #288, #291

### CODE REVIEW

File: server/poll/metadata.go [hunk 1]
```diff
@@ -2,18 +2,20 @@ package poll
 
 // Metadata stores personalized metadata of a poll.
 type Metadata struct {
-	PollID          string   `json:"poll_id"`
-	UserID          string   `json:"user_id"`
-	AdminPermission bool     `json:"admin_permission"` // AdminPermission will be true if the user with "UserID" has admin permission for the poll with "PollID", otherwise false.
-	VotedAnswers    []string `json:"voted_answers"`    // VotedAnswers is list of answer that the user with "UserID" have voted for the poll with "PollID"
+	VotedAnswers           []string `json:"voted_answers"` // VotedAnswers is list of answer that the user with "UserID" have voted for the poll with "PollID"
+	PollID                 string   `json:"poll_id"`
+	UserID                 string   `json:"user_id"`
```
Comment (hanzei) [inline server/poll/metadata.go:7]: Out of scope of this PR: Could you please elaborate why `userID` is needed here?

File: server/poll/metadata.go [hunk 2]
```diff
@@ -2,18 +2,20 @@ package poll
 
 // Metadata stores personalized metadata of a poll.
 type Metadata struct {
```
Comment (hanzei) [inline server/poll/metadata.go:4]: 0/5 thought, which it out of scope.: It feels like `Metadata` and `Poll` are very similar. Maybe they can be unified at some point.

File: server/poll/metadata.go (hunk 1 above)
Comment (kaakaa) [reply server/poll/metadata.go:7]: There is no practical reason, but I think that `pollID` and `userID` are primary key for `Metadata`.

File: server/poll/metadata.go (hunk 2 above)
Comment (kaakaa) [reply server/poll/metadata.go:4]: I thought the same thing, and I tried to add `poll.Settings` field to `Metadata`. But the only data types that can be handled between the server and webapp are those specified by gob [here](https://github.com/mattermost/mattermost-server/blob/e4fb5791b0ccfe22d2e2c8bdb7df5a410675f0a7/plugin/client_rpc.go#L95), so specifying the `poll.Settings` field resulted in an encoding error. As a results, we may not be able to use `Poll` type as `Metadata`.

File: server/poll/metadata.go (hunk 1 above)
Comment (hanzei) [reply server/poll/metadata.go:7]: It makes sense that `pollID` and `userID` are the primary key when fetching `Metadata`, but I'm wondering why they are also get returned? A client should how the `pollId` and `userID` itself.

**Pull request merged**